script/GameObjects/Jobs/Mine.lua

```diff
@@ -68,40 +68,41 @@ function Mine:getTravelTime (e)
 end
 
 function Mine:onUpdateActive (e, dt)
-  if not e.jobState then e.jobState = 0 end
-  e.jobState = e.jobState + 1
+  if not Config.game.gamePaused then
+    if not e.jobState then e.jobState = 0 end
```

Comment:
how about creating an enum for jobState

Reply:
[This comment will apply to all the "enum"-related change requests] Several days ago, I spent hours trying to work out a common way to do enums in the LTR code because enums are obviously the Right Way to handle well-defined modal values. Guess what? Lua has no built-in enum type or facility. Josh implemented one, but it seems to be specific to his core/phoenix interface code; it's not obviously used anywhere in the normal gameplay feature scripting code. So after spending hours fighting this, I moved on to a stupid initial implementation of menu state and entity types. (The eJobState pattern in the Mine and Transport jobs was created by Josh.) I agree enums would be the best way to handle cases like these. The question is whether it's important enough that it must be done Right Now, or if it's more important to finish fleshing out the basic systems then come back to do a cleanup pass and some refactoring once we see how all the basic gameplay code works together.

Reply:
define a global table somewhere e.g.:

```lua
Enums = {
	JobStates = {
		Docking = 1,
		LookingForTrades = 2,
		Trading = 3,
		WaitingForDelivery = 4,
		FinishedTrade = 5,
		Undocking = 6
	},
	OtherEnumClass = {
		OtherEnum = 1
	}
	-- go on
}
```

-- in Core.lua
require `PathToEnumFile`

should be as easy as that

Reply:
I've created a new Issue for converting existing (and these) cases of associated values into the Settings.addEnum() model that Josh provides. If possible, I'd like to request the current changes be accepted so I can resolve the new Issue by doing all the conversions to "enums" at the same time -- doing them together will same project time overall versus me trying to do each of these as a one-off right now.

Reply:
if you don´t want one big enum file just create an Enums Folder in root:

Enums/
Enums/Jobs.lua
Enums/Menu.lua
Enums/__init__.lua

__init__.lua:
```
require "Jobs"
require "Menu"
--etc
```

then in Core.lua
Enums = requireAll(Enums)